File: src/banks.rb
require 'json'

module Eav93
  module Sbp
    DATA_PATH = File.expand_path('../../data/banks.json', __FILE__)

    Bank = Struct.new(
      :id, :name, :logo, :schema, :package,
      :is_dr_active, :is_web_client_active, :web_client_url,
      keyword_init: true
    )

    Meta = Struct.new(:version, :updated_at, keyword_init: true)

    @cache = nil

    def self.get_banks
      _load['banks'].map do |item|
        Bank.new(
          id: item['id'],
          name: item['name'],
          logo: item['logo'],
          schema: item['schema'],
          package: item['package'],
          is_dr_active: item['isDrActive'] || false,
          is_web_client_active: item['isWebClientActive'] || false,
          web_client_url: item['webClientUrl']
        )
      end
    end

    def self.get_bank_by_id(id)
      get_banks.find { |b| b.id == id }
    end

    def self.get_meta
      data = _load
      Meta.new(version: data['version'], updated_at: data['updatedAt'])
    end

    def self._load
      @cache ||= JSON.parse(File.read(DATA_PATH))
    end
    private_class_method :_load
  end
end

File: src/banks.py
from __future__ import annotations

import json
from dataclasses import dataclass
from pathlib import Path
from typing import Optional

_DATA_PATH = Path(__file__).parent.parent / "data" / "banks.json"
_cache: dict | None = None


@dataclass(frozen=True)
class Bank:
    id: str
    name: str
    logo: Optional[str]
    schema: Optional[str]
    package: Optional[str]
    is_dr_active: bool
    is_web_client_active: bool
    web_client_url: Optional[str]


@dataclass(frozen=True)
class Meta:
    version: str
    updated_at: str


def _load() -> dict:
    global _cache
    if _cache is None:
        _cache = json.loads(_DATA_PATH.read_text(encoding="utf-8"))
    return _cache


def get_banks() -> list[Bank]:
    return [
        Bank(
            id=item["id"],
            name=item["name"],
            logo=item.get("logo"),
            schema=item.get("schema"),
            package=item.get("package"),
            is_dr_active=item.get("isDrActive", False),
            is_web_client_active=item.get("isWebClientActive", False),
            web_client_url=item.get("webClientUrl"),
        )
        for item in _load()["banks"]
    ]


def get_bank_by_id(bank_id: str) -> Optional[Bank]:
    return next((b for b in get_banks() if b.id == bank_id), None)


def get_meta() -> Meta:
    data = _load()
    return Meta(version=data["version"], updated_at=data["updatedAt"])
